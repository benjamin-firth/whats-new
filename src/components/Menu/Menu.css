.navBar {
  /* background-color: #D4E9D9; */
  border-right: 5px solid #D4E9D9;
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 17%;
}



