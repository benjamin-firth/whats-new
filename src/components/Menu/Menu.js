import React from 'react';
import './Menu.css'

const Menu = ({ clickHandler }) => {
  return (
    <nav className='navBar'>
      <button id='science' onClick={event => clickHandler(event)}>Science</button>
      <button id='entertainment' onClick={event => clickHandler(event)}>Entertainment</button>
      <button id='health' onClick={event => clickHandler(event)}>Health</button>
      <button id='technology' onClick={event => clickHandler(event)}>Technology</button>
      <button id='local' onClick={event => clickHandler(event)}>Local</button>
    </nav>
  )
}

export default Menu;