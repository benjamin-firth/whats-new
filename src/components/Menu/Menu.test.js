import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {
  let wrapper;
  let mockClickHandler;

  beforeEach(() => {
    mockClickHandler = jest.fn();
    wrapper = shallow(<Menu 
      clickHandler={mockClickHandler}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call the clickHandler prop method when button is clicked', () => {

    wrapper.find('button').at(0).simulate('click');
    expect(mockClickHandler).toHaveBeenCalled();

    wrapper.find('button').at(1).simulate('click');
    expect(mockClickHandler).toHaveBeenCalled();

    wrapper.find('button').at(2).simulate('click');
    expect(mockClickHandler).toHaveBeenCalled();

    wrapper.find('button').at(3).simulate('click');
    expect(mockClickHandler).toHaveBeenCalled();

    wrapper.find('button').at(4).simulate('click');
    expect(mockClickHandler).toHaveBeenCalled();

  })
})