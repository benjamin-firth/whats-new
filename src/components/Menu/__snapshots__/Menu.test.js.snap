// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot 1`] = `
<nav
  className="navBar"
>
  <button
    id="science"
    onClick={[MockFunction]}
  >
    Science
  </button>
  <button
    id="entertainment"
    onClick={[MockFunction]}
  >
    Entertainment
  </button>
  <button
    id="health"
    onClick={[MockFunction]}
  >
    Health
  </button>
  <button
    id="technology"
    onClick={[MockFunction]}
  >
    Technology
  </button>
  <button
    id="local"
    onClick={[MockFunction]}
  >
    Local
  </button>
</nav>
`;
